src/GitVersionHelper.php

```diff
@@ -63,4 +63,38 @@ public static function getNameAndVersion()
     {
         return self::appName() . '/' . self::getVersion();
     }
+    
+    /**
+     * Return commit hash with custom length
+     *
+     * @param null $length
+     * @throws CouldNotGetCommitHashException if there is no commit hash and `git
+     * rev-parse HEAD` fails
+     * @return string
+     *
+     */
+    public static function getHash($length=null)
+    {
+        // Remember current directory
+        $dir = getcwd();
+
+        // Change to base directory
+        chdir(base_path());
+
+        // Get version string from git
+        $commit = shell_exec('git rev-parse HEAD');
+
+        // Change back
+        chdir($dir);
+
+        if ($commit === null) {
+            throw new Exception\CouldNotGetCommitHashException;
+        }
+
+        if($length) {
+            $commit = substr($commit, 0, $length);
```

Comment:
Having thought about it again, I would much prefer using `git-rev-parse`'s `--short=NUM` option rather than using `substr`.